sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/JoinBenchmark.scala

```diff
@@ -178,6 +191,7 @@ object JoinBenchmark extends SqlBasedBenchmark {
       sortMergeJoin()
       sortMergeJoinWithDuplicates()
       shuffleHashJoin()
+      broadcastNestedLoopJoin()
```

Comment:
Could we update the benchmark results([JoinBenchmark-results.txt](https://github.com/apache/spark/blob/f5118f81e395bde0cd8253dbef6a9e6455c3958a/sql/core/benchmarks/JoinBenchmark-results.txt) and [JoinBenchmark-jdk11-results.txt](https://github.com/apache/spark/blob/f5118f81e395bde0cd8253dbef6a9e6455c3958a/sql/core/benchmarks/JoinBenchmark-jdk11-results.txt))?

Generate result:
```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.JoinBenchmark"
```

Reply:
@wangyum - sure, I noticed the original benchmark was running on `OpenJDK 64-Bit Server VM 1.8.0_232-8u232-b09-0ubuntu1~18.04.1-b09 on Linux 4.15.0-1044-aws` in [result file](https://github.com/apache/spark/blob/f5118f81e395bde0cd8253dbef6a9e6455c3958a/sql/core/benchmarks/JoinBenchmark-results.txt#L5). Should I try to run benchmark in the same type of machine? or this is not a hard requirement. cc @maropu as well.

Reply:
@wangyum, @maropu - updated the join benchmark results with same type machines as previous PRs. The description of PR has more details, thanks.